private_sharing/api_views.py

```diff
@@ -428,3 +441,23 @@ def post(self, request):
                 data_file.delete()
 
         return Response({"ids": ids}, status=status.HTTP_200_OK)
+
+
+class ListDataTypesView(ListAPIView):
+    """
+    Lists the datatypes available and which projects use them.
+    """
+
+    serializer_class = DataTypeSerializer
+
+    def get_queryset(self):
+        """
+        Get the queryset and filter on project if provided.
+        """
+        project_label = self.request.GET.get("project", None)
```

Comment:
should distinguish between source project and requesting project (this is source)